File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    N = brown +yellow
    for i in range(3, N):
        if N%i !=0:
            continue
            
        row = max(i,N//i)
        col = min(i,N//i)
        if (col+row)*2 -4 == brown and (col-2)*(row-2) == yellow:
            return[row,col]
            

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    for(int i=1;i<=yellow;i++){
        if(yellow%i == 0){
            int j = yellow / i;
            cout << i<<j<<"\n";
            if(2*(i+j+4)-4 == brown){
                answer = {max(i,j)+2,min(i,j)+2};
            }
        }
    }
    return answer;
}